### PR TITLE
rpc, wallet: add ability to retrieve all address book entries

### PR DESCRIPTION
**Pull request by w0xlt**
As far as I know, there is currently no way to retrieve all address book entries via RPC.
This functionality can be added by slightly modifying the `getaddressesbylabel` and making the `label` parameter optional.

PR:
```bash
./src/bitcoin-cli -regtest getaddressesbylabel
{
  "bcrt1q23d9qmeqrmdfazp79e5xavpgrzqmmtjf7k2p7u": {
    "label": "label03",
    "purpose": "receive"
  },
  "bcrt1pynqgl3mh8qmwj496nfg6ucxd8jw0wz3rz7jtf4s4v6e9zv6a4cnqdcl79k": {
    "label": "label02",
    "purpose": "receive"
  },
  "bcrt1pty99hxsx7trd53ul7lhl4ee82ggqslyq54mg5mte5wv8nfu7xxfqcf0pzz": {
    "label": "label01",
    "purpose": "receive"
  }
}
```

On master branch, it is not possible to run that RPC without passing a label parameter.

### CODE REVIEW

File: src/wallet/rpc/addresses.cpp [hunk 1]
```diff
@@ -640,14 +640,15 @@ RPCHelpMan getaddressesbylabel()
     return RPCHelpMan{"getaddressesbylabel",
                 "\nReturns the list of addresses assigned the specified label.\n",
```
Comment (jarolrod) [inline src/wallet/rpc/addresses.cpp:641]: If you want to do this, you should mention that if no label is supplied then all address book entry addresses are returned

Comment (w0xlt) [reply src/wallet/rpc/addresses.cpp]: Done in 37f809aad549368911b8358c1c6e6cb0e8c5bc7e. Thanks.

**Comment by luke-jr** [comment]
Concept ~0. Seems like a hack.

Code looks fine though.

**Comment by aureleoules** [comment]
Concept ACK but not sure about the approach. Would it be better to create a new RPC `getaddresses` with filters such as `label`, `purpose`, `creation_date`, `watch_only`, `minconf`, etc? Which would deprecate `getreceivedbylabel`.

**Comment by aureleoules** [comment]
Another approach would be to take a label `*` to list all addresses. This is a hack but it's already used for backwards compatibility in `listtransactions`.

**Comment by rajarshimaitra** [comment]
Concept ACK. This is useful for various situations where a bunch of imported spks needs to be retrieved from the wallet. Maybe to compare with an external source and check if any new spks need to be imported into the watchlist. Previously it could be done by making redundant labels just for this call, but having this option to dump everything seems useful.

I will go through the code changes in more detail. Adding one simple, functional test to check the behavior will be helpful.

**Comment by 1440000bytes** [comment]
Concept ACK

**Comment by achow101** [comment]
The PR didn't seem to attract much attention in the past. Also, the issue seems not important enough right now to keep it sitting around idle in the list of open PRs.

Closing due to lack of interest.

**Pull request closed without merging**